Update Unit Testing for Rotated Keybase Key

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

As part of https://discuss.hashicorp.com/t/hcsec-2021-12-codecov-security-event-and-hashicorp-gpg-key-exposure/23512, the Keybase `hashicorp` keys were rotated. Some testing was relying on the previous fingerprints, which can be seen with the unit tests here:

```
--- FAIL: TestFetchKeybasePubkeys (0.12s)
    keybase_test.go:40: fingerprints do not match; expected 
        []string{"0f801f518ec853daff611e836528efcac6caa3db", "91a6e7f85d05c65630bef18951852d87348ffc4c"}
        got
        []string{"0f801f518ec853daff611e836528efcac6caa3db", "c874011f0ab405110d02105534365d9472d7468f"}
```

It does not appear the IAM Access Key and User Login Profile acceptance testing is affected:

```
--- PASS: TestAccAWSUserLoginProfile_keybaseDoesntExist (10.31s)
--- PASS: TestAccAWSUserLoginProfile_notAKey (10.67s)
--- PASS: TestAccAWSAccessKey_encrypted (16.16s)
--- PASS: TestAccAWSAccessKey_basic (16.64s)
--- PASS: TestAccAWSUserLoginProfile_keybase (16.81s)
--- PASS: TestAccAWSUserLoginProfile_PasswordLength (17.00s)
--- PASS: TestAccAWSUserLoginProfile_basic (23.11s)
--- PASS: TestAccAWSAccessKey_Status (30.26s)
```

--- a/aws/internal/vault/helper/pgpkeys/keybase_test.go
+++ b/aws/internal/vault/helper/pgpkeys/keybase_test.go
@@ -33,7 +33,7 @@ func TestFetchKeybasePubkeys(t *testing.T) {
 
 	exp := []string{
 		"0f801f518ec853daff611e836528efcac6caa3db",
-		"91a6e7f85d05c65630bef18951852d87348ffc4c",
+		"c874011f0ab405110d02105534365d9472d7468f",
 	}
 
 	if !reflect.DeepEqual(fingerprints, exp) {

